assets/js/components/user-input/UserInputQuestionnaire.js

```diff
@@ -181,15 +182,15 @@ export default function UserInputQuestionnaire() {
 
 		const response = await saveUserInputSettings();
 		if ( ! response.error ) {
-			// If selected purpose has any ACR KMW assigned to it,
-			// mark 'includeConversionTailoredMetrics' key metrics setting to true
-			// so they can be included.
-			if ( haveConversionReportingEventsForTailoredMetrics ) {
-				setKeyMetricsSetting(
+			if ( isConversionReportingEnabled ) {
+				// Update 'includeConversionTailoredMetrics' key metrics setting with included
+				//conversion events, to mark that their respective metrics should be included in the
+				// list of tailored metrics and persist on the dashboard in case events are lost.
+				await setKeyMetricsSetting(
 					'includeConversionTailoredMetrics',
-					true
+					userInputPurposeConversionEvents
 				);
-				saveKeyMetricsSettings( {
+				await saveKeyMetricsSettings( {
```

Comment:
Await is added here and in above line to ensure saving of the setting happens before redirect